deploy/shakenfist_ci/tests/test_floating_ips.py

```diff
@@ -65,20 +60,27 @@ def test_simple(self):
 
         time.sleep(120)
 
+        console = base.LoggingSocket(inst['node'], inst['console_port'])
+        out = console.execute('cat /var/www/html/index.html')
+        if not out.find('Floating IPs work!'):
+            self.fail('User data script did not run!\n\n%s' % out)
+
         attempts = 0
         for _ in range(10):
             attempts += 1
             try:
                 r = requests.request(
                     'GET', 'http://%s/' % ifaces[0]['floating'])
-                self.assertEqual(200, r.status_code)
-                self.assertEqual('Floating IPs work!\n', r.text)
-                return
-
-            except Exception as e:
-                if attempts < 5:
-                    pass
-                else:
-                    raise e
+                success = 200 == r.status_code
```

Comment:
So you left this here to see if I'm reviewing?

Reply:
No, that's deliberate. It stores the value of the comparison of 200 and r.status_code in a variable called success, which is checked in the next line. Or am I missing something?

Reply:
Would it be simpler to write:
```
if r.status_code == 200 and r.text.find('Floating IPs work!') != -1:
    return
```

Reply:
He's a witch!

Reply:
Also yes.